Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1080,6 +1080,11 @@ theorem smul_eq_diagonal_mul [Fintype m] [DecidableEq m] (M : Matrix m n α) (a
   simp
 #align matrix.smul_eq_diagonal_mul Matrix.smul_eq_diagonal_mul
```

Comment:
It seems from the changed `simp` calls that `simp` should really be able to identify `a • M = (diagonal fun _ => a) * M`. But I'm not sure which in which direction we'd want this equality, or maybe even something slightly different than the current statement. Or is this better for a follow-up?

Reply:
I was hoping to leave this for a follow-up because I didn't want to have to think about it 🙃. I think generally the `smul` version is probably more convenient as the simp normal form, which makes it easier to move `a` around, but harder to associate it.